plugins/foreman_maintain/0.2/index.md

```diff
@@ -0,0 +1,405 @@
+
+For general information about foreman-maintain, please check the [github repo](https://github.com/theforeman/foreman_maintain)
+
+# Install
+
+To install foreman-maintain on foreman or katello instances, please run:
+```
+# yum install rubygem-foreman_maintain
+```
+
+Foreman-maintain is installed as a dependency in Katello 3.7 and up.
+
+# Service management
+
+Services can be managed using `foreman-maintain service`. This is based on the existing `katello-service` command. For all options, run `foreman-maintain service --help`.
+
+# Backup
+
+### Supported systems
+Backup and Restore in foreman-maintain works on the following:
+* Katello (>3.0)
+* Foreman-proxy with content (>3.0)
+
+Backup and Restore support for standalone Foreman is currently in development.
+
+
+### General
+Note: We assume all the foreman-maintain commands are run as a root user.
+
+Backup utility in `foreman-maintain` offers three backup strategies
+* Offline - The whole Katello instance will be turned off completely for the entire backup.
+* Online - Backing up the repositories can take an extensive amount of time. You can perform a backup while online. In order for this procedure to succeed, you must not change or update the repositories database until the backup procedure is complete. Thus, you must avoid publishing, adding, or deleting content views, promoting content view versions, adding, changing, or deleting sync-plans, and adding, deleting, or syncing repositories during this time.
+* Snapshot - The whole Katello instance will be turned off only for time necessary to create and mount logical volume snapshots. The backup is done from the snapshots after the instance is turned on again. This allows to minimize the maintenance window necessary for doing the backup.
+
+### Incremental backup
+Incremental backups can be used to only store the changes since the last backup:
+
+First take a full backup (it is applicable for all strategies `offline` will be used in the examples):
+```
+# foreman-maintain backup offline /tmp/backups
+```
+(This will create a new directory, /tmp/backups/katello-backup-YYYY-MM-DD-hh-mm-ss)
+
+Take 1st incremental backup (this will create a new directory under /tmp/backups to house the new backup, just like the full backup directory):
+
+```
+# foreman-maintain backup offline --incremental /tmp/backup/FULL_BACKUP_DIR /tmp/backups
+```
+
+Take 2nd incremental backup (again, this will create a new directory under /tmp/backups to house this second incremental backup):
+```
+# foreman-maintain backup offline --incremental /tmp/backup/FIRST_INCREMENTAL_BACKUP_DIR /tmp/backups
+```
+
+#### Branching/rebasing incremental backups
+
+Should you choose to take a new incremental backup from, say, the full backup so you don’t need too many files when restoring those backups, simply point the command to the full backup directory, and this newest backup directory will be incremental in relation to the full backup not to the 2nd incremental backup:
```

Comment:
This is a bit of a run on sentence that I think could flow nicer if it was split.